server.go

```diff
@@ -147,14 +166,37 @@ func (server *Server) ID() peer.ID {
 
 func (server *Server) handle(s *streamWrap) error {
 	logger.Debugf("%s: handling remote RPC", server.host.ID().Pretty())
+	var err error
 	var svcID ServiceID
 	var argv, replyv reflect.Value
+	ctx := context.Background()
 
-	err := s.dec.Decode(&svcID)
+	err = s.dec.Decode(&svcID)
 	if err != nil {
 		return newServerError(err)
 	}
 
+	sh := server.statsHandler
+	if sh != nil {
```

Comment:
This section seems very similar to the one in the client. Is it possible to have a helper method for both?

Reply:
They differ in the stats that get initialized and they will diverge further as I get the `OutPayload` and `InPayload` stuff fleshed with respect to payload size/length etc.